src/vaults/yields.utils.ts

```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
+  let totalAccumulated = 0;
+  let totalVaultTokens = 0;
+  let totalVaultPrincipal = 0;
+
+  console.log(`Evaluating ${allEvents.length} events for ${vault.name}`);
+  for (const event of allEvents) {
+    const token = await getFullToken(chain, event.token);
+    const amount = formatBalance(event.amount, token.decimals);
+    const { price } = await queryPriceAtTimestamp(token.address, event.timestamp * 1000);
+
+    const tokenEarned = price * amount;
+    totalAccumulated += tokenEarned;
+    if (event.type === HarvestType.Harvest) {
+      totalHarvested += amount;
+    }
+
+    let balance = 0;
+    if (isInfluenceVault(vault.address)) {
+      balance = await getInfuelnceVaultYieldBalance(chain, vault.address, event.block);
+    } else {
+      const vaultContract = Vault__factory.connect(vault.address, sdk.provider);
+      const totalSupply = await vaultContract.totalSupply({ blockTag: event.block });
+      balance = formatBalance(totalSupply);
+    }
+    const { price: vaultPrice } = await queryPriceAtTimestamp(vault.address, event.timestamp * 1000);
+    totalVaultTokens += balance;
+    totalVaultPrincipal += vaultPrice * balance;
+
+    harvestReport = harvestReport.concat(
+      `[${new Date(event.timestamp * 1000).toDateString()}] ${amount.toFixed(2)} ${
+        token.symbol
+      } ($${tokenEarned.toFixed(2)}) - Balance: ${balance.toFixed(2)}\n`,
+    );
+  }
+
+  const duration = VAULT_TWAY_PERIOD * ONE_DAY_MS;
+  const averagePrincipal = totalVaultPrincipal / allEvents.length;
+  const averageTokens = totalVaultTokens / allEvents.length;
+  const apr = calculateYield(averagePrincipal, totalAccumulated, duration);
+  harvestReport = harvestReport.concat(
+    `Vault Holdings: $${averagePrincipal.toFixed()} (${averageTokens.toFixed()} tokens), Total Earned: $${totalAccumulated.toFixed(
+      2,
+    )}, Est. Yield: ${apr.toFixed(2)}%`,
+  );
+  console.log(harvestReport);
+
+  const valueSources = [];
+
+  // create the apr source for harvests
+  const compoundApr = calculateYield(totalVaultTokens, totalHarvested, duration);
+  const compoundYieldSource = createYieldSource(vault, SourceType.PreCompound, VAULT_SOURCE, compoundApr);
+  valueSources.push(compoundYieldSource);
+
+  // create the apy source for harvests
+  const compoundApy = calculateYield(totalVaultTokens, totalHarvested, duration, totalHarvested);
+  const compoundedYieldSource = createYieldSource(vault, SourceType.Compound, VAULT_SOURCE, compoundApy);
+  valueSources.push(compoundedYieldSource);
+
+  const tokensEmitted = new Map<string, BigNumber>();
+  for (const distribution of allDistributions) {
+    const { token, amount } = distribution;
+    let entry = tokensEmitted.get(token);
+    if (!entry) {
+      entry = BigNumber.from(0);
+      tokensEmitted.set(token, entry);
+    }
+    tokensEmitted.set(token, entry.add(amount));
+  }
+
+  let flywheelCompounding = 0;
+
+  for (const [token, amount] of tokensEmitted.entries()) {
```

Comment:
this is old code, will suffice for now - likely need to do another round to utilize the captures above vs. this as we are losing time granularity here (and thus subject to price fluctuation inaccuracies)